src/network/networkd-network.c

```diff
@@ -165,7 +165,7 @@ static int network_load_one(Manager *manager, const char *filename) {
                 return r;
 
         LIST_FOREACH(routes, route, network->static_routes) {
-                if (!route->family) {
+                if (!route->family && !route->dev) {
```

Comment:
btw, isn't "route-family == AF_UNSPEC" a nicer check for this? that's at least what you use elsewhere...